pkg/daemon/update.go

```diff
@@ -1988,6 +1990,77 @@ func (dn *Daemon) cancelSIGTERM() {
 	}
 }
 
+func (dn *Daemon) SetLabelsOnNode(labels map[string]interface{}) error {
+	var err error
+	var patchData []byte
+	patch := struct {
+		Metadata map[string]interface{} `json:"metadata"`
+	}{
+		Metadata: map[string]interface{}{
+			"labels": labels,
+		},
+	}
+	dn.logSystem("Setting labels %v on node %s", labels, dn.name)
+	patchData, err = json.Marshal(&patch)
+	if err != nil {
+		dn.logSystem("Error in setting labels on node %s: %v", dn.name, err)
+		return err
+	}
+	_, err = dn.kubeClient.CoreV1().Nodes().Patch(
+		context.TODO(), dn.name, types.MergePatchType, patchData, metav1.PatchOptions{})
+	if err != nil {
+		dn.logSystem("Error setting labels on node %s: %v", dn.name, err)
+		return err
+	}
+	return nil
+}
+
+func (dn *Daemon) setExcludeFromLBLabel(addLabel bool) error {
+	var labelValue interface{}
+	var err error
+	if addLabel {
+		labelValue = ""
+	}
+	dn.logSystem("Patching node %s with label %s=%s (addLabel=%t)",
+		dn.name, excludeFromLBLabel, labelValue, addLabel)
+	err = dn.SetLabelsOnNode(map[string]interface{}{excludeFromLBLabel: labelValue})
+	if err != nil {
+		dn.logSystem("Error patching node %s with label %s=%s: %v",
+			dn.name, excludeFromLBLabel, labelValue, err)
+		return err
+	}
+	return nil
+}
+
+func (dn *Daemon) addExcludeFromLBLabel() error {
+	dn.setExcludeFromLBLabel(true)
+	// adding dummy file to mark that the label was added by MCD
+	dummyfile, err := os.Create(excludeExtLBPath)
+	if err != nil {
+		dn.logSystem("Error creating file %s: %v", excludeExtLBPath, err)
+		return err
+	}
+	dummyfile.Close()
+	return nil
+}
+
+func (dn *Daemon) removeExcludeFromLBLabel() error {
+	var err error
+	// checking for dummy file marking that we previously added the label
+	_, err = os.Stat(excludeExtLBPath)
+	// file does not exist, the label (if any) was not added by MCD, keep it
+	if os.IsNotExist(err) {
+		dn.logSystem("File %s not present: nothing to do", excludeExtLBPath)
+		return nil
+	}
+	err = dn.setExcludeFromLBLabel(false)
```

Comment:
Can you explain what this is doing? Why would we call the set function from remove? Can we do this without the bool usage?